libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/KeyStoreUtil.java

```diff
@@ -148,5 +159,90 @@ static X509ExtendedTrustManager createTrustManager(@Nullable KeyStore trustStore
             + "] and truststore [" + trustStore + "]");
     }
 
+    /**
+     * Creates a {@link X509ExtendedTrustManager} based on the provided certificates
+     *
+     * @param certificates the certificates to trust
+     * @return a trust manager that trusts the provided certificates
+     */
+    public static X509ExtendedTrustManager createTrustManager(Collection<Certificate> certificates) throws GeneralSecurityException {
+        KeyStore store = buildTrustStore(certificates);
+        return createTrustManager(store, TrustManagerFactory.getDefaultAlgorithm());
+    }
+
+    static Stream<KeyStoreEntry> stream(KeyStore keyStore,
+                                        Function<GeneralSecurityException, ? extends RuntimeException> exceptionHandler) {
+        try {
+            return Collections.list(keyStore.aliases()).stream().map(a -> new KeyStoreEntry(keyStore, a, exceptionHandler));
+        } catch (KeyStoreException e) {
+            throw exceptionHandler.apply(e);
+        }
+    }
+
+    static class KeyStoreEntry {
+        private final KeyStore store;
+        private final String alias;
+        private final Function<GeneralSecurityException, ? extends RuntimeException> exceptionHandler;
+
+        KeyStoreEntry(KeyStore store, String alias, Function<GeneralSecurityException, ? extends RuntimeException> exceptionHandler) {
+            this.store = store;
+            this.alias = alias;
+            this.exceptionHandler = exceptionHandler;
+        }
+
+        public String getAlias() {
+            return alias;
+        }
+
+        public X509Certificate getX509Certificate() {
+            try {
+                final Certificate c = store.getCertificate(alias);
+                if (c instanceof X509Certificate) {
+                    return (X509Certificate) c;
+                } else {
+                    return null;
+                }
+            } catch (KeyStoreException e) {
+                throw exceptionHandler.apply(e);
+            }
+        }
+
+        public boolean isKeyEntry() {
+            try {
+                return store.isKeyEntry(alias);
+            } catch (KeyStoreException e) {
+                throw exceptionHandler.apply(e);
+            }
+        }
+
+        public PrivateKey getKey(char[] password) {
+            try {
+                final Key key = store.getKey(alias, password);
+                if (key instanceof PrivateKey) {
+                    return (PrivateKey) key;
+                }
+                return null;
+            } catch (GeneralSecurityException e) {
+                throw exceptionHandler.apply(e);
+            }
+        }
+
+        public List<? extends X509Certificate> getX509CertificateChain() {
+            try {
+                final Certificate[] certificates = store.getCertificateChain(alias);
+                if (certificates == null || certificates.length == 0) {
+                    return List.of();
```

Comment:
No, it is intentional that this returns a more useful API than `Keystore`.
I can document it if you'd like, but I feel pretty strongly that the purpose writing this wrapper is to avoid those cumbersome low level details from `Keystore`.

Reply:
No need to document additionally I think, reasoning makes sense to me